test/TestERC20_Transfer.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity >=0.5.16;
+
+import "./firefly/FireflyCheck.sol";
+import "../contracts/ERC20.sol";
+
+contract TestERC20Firefly_Transfer is FireflyCheck {
+    function _testTransfer() public {
+        address owner = this.firefly_genAddress();
+        uint256 initSupply = this.firefly_genUint256();
+
+        ERC20 token = new ERC20("gold", "GLD", owner, initSupply);
+
+        address from  = address(this);
+        address to    = this.firefly_genAddress();
+        uint256 value = this.firefly_genUint256();
+
+        require(from != to);
+
+        uint256 balFrom = token.balanceOf(from);
+        uint256 balTo   = token.balanceOf(to);
+        require(value <= balFrom);
+
+        require(token.transfer(to, value));
+
+        uint256 balFromAfter = token.balanceOf(from);
+        uint256 balToAfter   = token.balanceOf(to);
+
+        assert(balFromAfter <= balFrom); //
+        assert(balToAfter >= balTo);     // No overflows
```

Comment:
I think this should be a `require`, not an `assert`, so that you're saying "assuming there is no overflow, the balances are updated correctly".

As it stands, I think the second `assert` here will have a failing branch always, because the prover will say "what if `balTo + value` overflows?"

Reply:
Aren't we not just assuming there's no overflow, but are trying to specify it? I assume you wouldn't want it to happen in your ERC20 token.